temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java

```diff
@@ -145,22 +145,50 @@ <R> R update(
    * @param firstExecutionRunId specifies the RunID expected to identify the first run in the
    *     workflow execution chain. If this expectation does not match then the server will reject
    *     the update request with an error.
+   * @param waitPolicy specifies at what point in the update request life cycles this request should
+   *     return.
    * @param resultClass class of the update return value.
    * @param <R> type of the update return value.
-   * @param resultType type of the workflow return value. Differs from resultClass for generic
-   *     types.
+   * @param resultType type of the update return value. Differs from resultClass for generic types.
    * @param args update method arguments
-   * @return update reference that can be used to get the result of the update.
+   * @return update handle that can be used to get the result of the update.
    */
   @Experimental
   <R> UpdateHandle<R> startUpdate(
       String updateName,
       String updateId,
       String firstExecutionRunId,
+      UpdateWorkflowExecutionLifecycleStage waitPolicy,
       Class<R> resultClass,
       Type resultType,
       Object... args);
 
+  /**
+   * Get an update handle to an update request previously started. Getting an update handle does not
+   * guarantee the update ID exists.
+   *
+   * @param updateId the identifier for the requested update.
+   * @param resultClass class of the update return value.
+   * @param <R> type of the update return value.
+   * @return update handle that can be used to get the result of the update.
+   */
+  @Experimental
+  <R> UpdateHandle<R> getUpdateHandle(String updateId, Class<R> resultClass);
```

Comment:
I wonder if this would make more sense as `newUpdateStub` and move start under there (but can still leave shortcuts out here). This matches other Java approaches, though would make it unlike other SDKs. Hrmm.

Reply:
I think the only other java approach this matches is what we do for workflow with the untyped stub? I strongly prefer the API of handle over stub personally so I would keep it as is.

Reply:
I agree, I think I like the handle approach (though probably less strongly than you). Just wanted to note that the other things that have start-then-complete semantics do use stubs (with shortcuts/helpers as needed).